### PR TITLE
Reference HZ by ID instead of name

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/12944

When running aws-operator in tenant cluster we discovered CF is confused
because we have both public and private HZ for the TC base domain. Because of
that HZ needs to referenced with the ID instead of the name to mane the
selection unambiguous.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by kopiczko** [comment]
I need to test TC upgrades still.

File: service/controller/resource/tccpf/create.go [hunk 1]
```diff
@@ -170,10 +170,11 @@ func (r *Resource) newRecordSetsParams(ctx context.Context, cr infrastructurev1a
 	var recordSets *template.ParamsMainRecordSets
 	{
 		recordSets = &template.ParamsMainRecordSets{
-			BaseDomain:                 key.ClusterBaseDomain(cr),
-			ClusterID:                  key.ClusterID(&cr),
-			GuestHostedZoneNameServers: cc.Status.TenantCluster.DNS.HostedZoneNameServers,
-			Route53Enabled:             r.route53Enabled,
+			BaseDomain:                     key.ClusterBaseDomain(cr),
+			ClusterID:                      key.ClusterID(&cr),
+			ControlPlanePublicHostedZoneID: cc.Status.ControlPlane.HostedZone.ID,
```
Comment (xh3b4sd) [inline service/controller/resource/tccpf/create.go:175]: When this is used here, should we have a check at the beginning of `EnsureCreated` to cancel if the ID is not yet present.

**Review by xh3b4sd — APPROVED** [review]

**Review by corest — APPROVED** [review]

File: service/controller/resource/tccpf/create.go (hunk 1 above)
Comment (kopiczko) [reply service/controller/resource/tccpf/create.go:183]: I just wrote the check and then realised it should be always available (this is CP HostedZone ID) unless Route53 is disabled (in which case we need to ignore empty value anyway).

**Comment by kopiczko** [comment]
Tested on ginger. NS servers remained the same after the upgrade.

**Review by njuettner — APPROVED** [review]

**Pull request merged**